kubernetes/core/src/main/kotlin/com/google/kubernetes/tools/core/KubectlCommandLineState.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.kubernetes.tools.core
+
+
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.Executor
+import com.intellij.execution.configurations.*
+import com.intellij.execution.process.KillableProcessHandler
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.options.SettingsEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.xmlb.XmlSerializer
+import org.jdom.Element
+import java.io.File
+import javax.swing.JComponent
+
+/**
+ * Runs Skaffold on command line in both "run"/"dev" modes based on the given
```

Comment:
Two comments here:
- first there is some cleanup items left (like "skaffold" references in the comments)
- generally you don't want to only copy the implementation from the skaffold version. first think about the goal here (i.e. to be able to execute a kubectl command and get the results). There may be some overlap, but the Skaffold implementation was more focused on executing the process and displaying it in a console window (see comment about `CommandLineState`), whereas here we aren't opening a console window. There is probably some overlap in the implementation, but in these cases we can extract out common code to avoid duplication.

Reply:
here is a super simplified example of using `ProcessBuilder` to execute a process and read the output:
https://stackoverflow.com/questions/14915319/get-output-of-terminal-command-using-java?lq=1